src/graphics/reproject-texture.js

```diff
@@ -48,15 +49,38 @@ function getProjectionName(projection) {
  * function can read and write textures with pixel data in RGBE, RGBM, linear and sRGB formats. When
  * specularPower is specified it will perform a phong-weighted convolution of the source (for generating
  * a gloss maps).
- * @param {GraphicsDevice} device - The graphics device.
  * @param {Texture} source - The source texture.
  * @param {Texture} target - The target texture.
- * @param {number} [specularPower] - Optional specular power. When specular power is specified,
+ * @param {object} [options] - The options object.
+ * @param {GraphicsDevice} [options.device] - Optional graphics device (default is the source texture device).
+ * @param {number} [options.specularPower] - Optional specular power. When specular power is specified,
  * the source is convolved by a phong-weighted kernel raised to the specified power. Otherwise
  * the function performs a standard resample.
- * @param {number} [numSamples] - Optional number of samples (default is 1024).
+ * @param {number} [options.numSamples] - Optional number of samples (default is 1024).
+ * @param {number} [options.face] - Optional cubemap face to update (default is update all faces).
  */
-function reprojectTexture(device, source, target, specularPower = 1, numSamples = 1024) {
+function reprojectTexture(source, target, options = {}) {
+    // maintain backwards compatibility with previous function signature
+    // reprojectTexture(device, source, target, specularPower = 1, numSamples = 1024)
+    if (source.constructor === GraphicsDevice) {
+        source = arguments[1];
+        target = arguments[2];
+        options = {
+            device: arguments[0],
+            specularPower: arguments[3] === undefined ? 1 : arguments[3],
+            numSamples: arguments[4] === undefined ? 1024 : arguments[4]
+        };
+        // #if _DEBUG
+        console.warn('DEPRECATED: please use the updated pc.reprojectTexture API.');
+        // #endif
+    }
+
+    // extract options
+    const device = options.device || source.device;
```

Comment:
I don't see how it's possible for `source.device` to be invalid.

Reply:
Correct. Question is whether it's necessary to keep supporting user provided device?

Reply:
probably remove options.device completely .. texture always has a valid device.

Reply:
I'll remove :+1: